### PR TITLE
RATEST-210:Refactor RefApp 3.x registrationsteps to reflect two scenarios

### PR DESCRIPTION
**Pull request by sherrif10**
https://issues.openmrs.org/browse/RATEST-210
What changed
For this workflow to reflect exactly what it handles, it should be considerate to be reflecting the exact class, so i refactored registration.js classand its feature file for future use and followup making it easy cc @jayasanka-sack @brandones @kdaud

### CODE REVIEW

**Comment by sherrif10** [comment]
cc @brandones   @jayasanka-sack  @kdaud . Looking forward for your reviews thanks

**Review by jayasanka-sack — CHANGES_REQUESTED** [review]
Thanks for getting this done @sharif. 

It seems the registration flow is not running after this change. Could you please check it?

https://github.com/openmrs/openmrs-contrib-qaframework/pull/98/checks?check_run_id=3470487437

File: package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
     "ocl": "./node_modules/.bin/cypress-tags run -eTAGS='@cypress and @ocl'",
     "microfrontend": "./node_modules/.bin/cypress-tags run -eTAGS='@cypress and @microfrontend'",
     "refapp3Login": "cypress run --spec src/test/resources/features/refapp-3.x/01-login/login.feature",
-    "refapp3Registration": "cypress run --spec src/test/resources/features/refapp-3.x/02-registration/registration.feature",
+    "refapp3searchAndRegistration": "cypress run --spec src/test/resources/features/refapp-3.x/02-registration/search-and-registration.feature",
```
Comment (jayasanka-sack) [inline package.json:23]:
```suggestion
    "refapp3searchAndRegistration": "cypress run --spec src/test/resources/features/refapp-3.x/02-search-and-registration/search-and-registration.feature",
```

Comment (sherrif10) [reply package.json]: Hello @jayasanka-sack , are you suggesting we remove that folder

**Comment by sherrif10** [comment]
Hello @jayasanka-sack @kdaud   , Am into this to get it merged. What is your opinion

File: package.json (hunk 1 above)
Comment (kdaud) [inline package.json:23]: Renaming the script to refapp3SearchAndRegistration follows other existing workflow names

**Comment by kdaud** [comment]
Worth looking into the build [failures](https://github.com/openmrs/openmrs-contrib-qaframework/pull/98/checks?check_run_id=3471235689)

**Comment by sherrif10** [comment]
Am looking into build failure @kdaud  .Thanks

**Comment by sherrif10** [comment]
Hey @jayasanka-sack @kdaud  , seems these failures are not about the commit.

File: package.json (hunk 1 above)
Comment (jayasanka-sack) [reply package.json]: No, @sherrif10 the correct command should be:

`cypress run --spec src/test/resources/features/refapp-3.x/02-search-and-registration/search-and-registration.feature`

File: package.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
     "ocl": "./node_modules/.bin/cypress-tags run -eTAGS='@cypress and @ocl'",
     "microfrontend": "./node_modules/.bin/cypress-tags run -eTAGS='@cypress and @microfrontend'",
     "refapp3Login": "cypress run --spec src/test/resources/features/refapp-3.x/01-login/login.feature",
-    "refapp3Registration": "cypress run --spec src/test/resources/features/refapp-3.x/02-registration/registration.feature",
+    "refapp3SearchAndRegistration": "cypress run --spec src/test/resources/features/refapp-3.x/02-registration/search-and-registration.feature",
```
Comment (jayasanka-sack) [inline package.json:23]: The issue is here. Please change this line to:

```suggestion
    "refapp3searchAndRegistration": "cypress run --spec src/test/resources/features/refapp-3.x/02-search-and-registration/search-and-registration.feature",
```

**Comment by jayasanka-sack** [comment]
Fixed the issue. 🤗

There were two issues.
1. There was a typo in the npm script's name
1. Provided path to the feature file was wrong.

**Review by jayasanka-sack — APPROVED** [review]
Thanks for getting this done @sherrif10 !

**Pull request merged**